cmd/thanos/receive.go

```diff
@@ -507,7 +564,28 @@ func setupHashring(g *run.Group,
 				if !ok {
 					return nil
 				}
+
+				for _, hc := range dbs.GetHashringConfig() {
```

Comment:
Let's do this logic inside multitsdb when we call `SetHashringConfig`

Reply:
Which logic? The log? I added another log after calling `SetHashringConfig` [here](https://github.com/thanos-io/thanos/pull/5777/files#diff-3d45f9beb5b670e94fe2715e21477a3fb8f5db7bf97f01cddf6e108bb98761a6R583-R587)

Reply:
These logs only serve showing the flow of change of external labels. Once the handling changes work I'll delete these logs

Reply:
You mean we should add the logs inside `SetHashringConfig`? I think adding the logs before and after calling `SetHashringConfig` is better. Or you mean something else that I misunderstood?

Reply:
Not the log, I meant we should only call `dbs. SetHashringConfig` from this function. Extracting external labels should be done inside that function.

Reply:
> Not the log, I meant we should only call `dbs. SetHashringConfig` from this function. Extracting external labels should be done inside that function.
Can you specify which function you pointing to? Right now your review's pointing to the log part.

Reply:
So the section that starts with ```
for _, hc := range dbs.GetHashringConfig()
```
should be done inside `dbs. SetHashringConfig`. 

Does this make sense?

Reply:
You mean [this section](https://github.com/thanos-io/thanos/blob/bf8f5007fbfc9bf67f93be1d1bdee596e8f1f9fe/cmd/thanos/receive.go#L568-L572)?

```
				for _, hc := range dbs.GetHashringConfig() {
					for _, el := range hc.ExternalLabels {
						level.Info(logger).Log("msg", "2. External label before reloading config: "+el)
					}
				}
```

That's the log.

Reply:
If you mean the log section, then back to my previous comments:

> You mean we should add the logs inside `SetHashringConfig`? I think adding the logs before and after calling `SetHashringConfig` is better.

> These logs only serve showing the flow of change of external labels. Once the changes handling works I'll delete these logs.